extensions/geode-modules/src/main/java/org/apache/geode/modules/session/catalina/DeltaSession.java

```diff
@@ -553,8 +555,30 @@ public void fromData(DataInput in) throws IOException, ClassNotFoundException {
     }
   }
 
-  protected Map readInAttributes(final DataInput in) throws IOException, ClassNotFoundException {
-    return DataSerializer.readObject(in);
+  private void readInAttributes(DataInput in) throws IOException, ClassNotFoundException {
+    Map map = DataSerializer.readObject(in);
+    ConcurrentMap newMap = new ConcurrentHashMap();
+    newMap.putAll(map);
+    try {
+      Field field = getAttributesFieldObject();
+      field.setAccessible(true);
+      field.set(this, newMap);
+    } catch (NoSuchFieldException e) {
+      logError(e);
```

Comment:
I'll throw NoSuchElementException and IllegalStateException if any of these occur